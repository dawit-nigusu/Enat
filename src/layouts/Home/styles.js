const React = require("react-native");
const {Dimensions} = React;
const deviceHeight = Dimensions.get("window").height;


export default {
    container: {
        backgroundColor: "#FFF"
    },
    mb10: {
        marginBottom: 10
    },
    text: {
        color: "#D8D8D8",
        bottom: 6,
        marginTop: 5
    },
    homeImage: {
        width: null,
        //height: deviceHeight * 0.45,
        flex: 1,
    },
    mr10: {
        marginRight: 10
    },
    tile: {
        backgroundColor: "rgba(245, 58, 124, 0.5)",
        marginLeft: 10
    },
    mediumText: {
        color: "#ffffff",
        textAlign: 'center',
        fontSize: 22,
        alignSelf: "center",
        padding: 5
    },
    largeText: {
        color: "#ffffff",
        textAlign: 'center',
        fontSize: 36,
        fontWeight: 'bold',
        alignSelf: "center"
    },
    bottom: {
        justifyContent: 'flex-end'
    },
    subTitleText: {
        color: "#ffffff",
        textAlign: 'center',
        fontSize: 16,
        alignSelf: "center",
        padding: 5,
    },
    titleText: {
        color: "#ffffff",
        textAlign: 'center',
        fontSize: 18,
        fontWeight: 'bold',
        alignSelf: "center"
    },

};
