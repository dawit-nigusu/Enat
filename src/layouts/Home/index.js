import React, {Component} from "react";
import {ImageBackground} from "react-native";
import {Container, Text, View} from "native-base";
import {Col, Row} from "react-native-easy-grid";
import styles from "./styles";
import Header from "../Common/Header";
import Footer from "../Common/Footer";
import variable from "./../../theme/variables/platform";
import moment from "moment/moment";
import {toEthiopian, toGregorian} from "../../Calendar";
import {retrieveItem, storeItem} from "../../Storage";
import * as FileSystem from 'expo-file-system';
import * as Permissions from 'expo-permissions';

const DUE_DATE = 'dueDate';
const cover = require('../../../assets/home.jpg');
const MONTH = ['መስከረም', 'ጥቅምት', 'ህዳር', 'ታህሳስ', 'ጥር', 'የካቲት',
    'መጋቢት', 'ሚያዚያ', 'ግንቦት', 'ሰኔ', 'ሐምሌ', 'ነሐሴ', 'ጳጉሜ'];

const DueDateBanner = (props) => {
    if (props.isBorn) {
        return (
            <Col style={{
                width: "100%",
                backgroundColor: "rgba(235, 58, 124, 0.5)",
                padding: 7,
            }}>
                <Text style={{
                    color: "white",
                    fontSize: 32,
                    textAlign: "center",
                    textAlignVertical: "center",
                    fontWeight: "bold",
                }}>ልጅሽ ተወልዷል</Text>
            </Col>
        );
    }
    return (
        <Col style={{
            width: "100%",
            backgroundColor: "rgba(235, 58, 124, 0.5)",
            padding: 7,
        }}>
            <Text style={[styles.titleText]}>አሁን ያለሽበት ሳምንት {props.currentWeek} ሳምንት</Text>
            <Text style={[styles.titleText]}>
                ልጅሽን ለመውለድ የቀረሽ {props.weeksRemaining} ሳምንት ከ{props.daysRemaining} ቀን
            </Text>
            <Text style={[styles.titleText]}>ቀኑ {props.birthday}</Text>
        </Col>
    );
};

class Home extends Component {

    constructor(props) {
        super(props);
        this.state = {
            weeksRemaining: 0,
            daysRemaining: 0,
            currentWeek: 0,
            day: "",
            month: "",
            year: "",
            birthday: "",
            isBorn: false,
            isReady: false,
        };
    }

    componentWillMount() {
        this._getDueDate();
        // try{
        //     FileSystem.makeDirectoryAsync(FileSystem.documentDirectory + 'enat').then(() => {
        //         try {
        //             FileSystem.downloadAsync(
        //                 'http://192.168.98.2:8000/media/test.png',
        //                 FileSystem.documentDirectory + "enat/banner.png"
        //             ).then(({uri}) => {
        //                 storeItem('banner_uri', uri).then(() => {
        //                     console.log('Finished downloading to ', uri);
        //                 }).catch((error) => {
        //                 });
        //             }).catch((error) => {
        //                 console.error(error);
        //             });
        //         }
        //         catch (error){
        //            console.log(error)
        //         }
        //
        //     }).catch(e => {
        //         console.log(e);
        //     });
        // }
        // catch (error)
        // {
        //      console.log(error)
        // }

    }

    _getDueDate() {
        let [y, m, d] = toEthiopian(moment().year(), moment().month() + 1, moment().date());

        retrieveItem(DUE_DATE).then((item) => {
            // console.log(item);
            if (item !== undefined && item != null) {
                if (item.date !== undefined && item.date !== null) {
                    this.setState({day: parseInt(item.date),});
                    if (item.date !== undefined && item.date !== null) {
                        this.setState({month: item.month,});
                        if (item.date !== undefined && item.date !== null) {
                            this.setState({
                                year: parseInt(item.year),
                                birthday: `${MONTH[item.month]} ${item.date}፤ ${item.year}`
                            })
                        } else {
                            this.setState({
                                day: d,
                                month: m - 1,
                                year: y,
                                birthday: `${MONTH[m - 1]} ${d}፤ ${y}`,
                            });
                        }
                    } else {
                        this.setState({
                            day: d,
                            month: m - 1,
                            year: y,
                            birthday: `${MONTH[m - 1]} ${d}፤ ${y}`,
                        });
                    }
                } else {
                    this.setState({
                        day: d,
                        month: m - 1,
                        year: y,
                        birthday: `${MONTH[m - 1]} ${d}፤ ${y}`,
                    });
                }
            } else {
                this.setState({
                    day: d,
                    month: m - 1,
                    year: y,
                    birthday: `${MONTH[m - 1]} ${d}፤ ${y}`,
                });
            }
            this.dueDateCalculator();
        }).catch(function (error) {
            console.error(error.message);
        });

    }

    dueDateCalculator() {
        let [y, m, d] = toGregorian(this.state.year, this.state.month + 1, this.state.day);
        let duration = moment([y, m - 1, d]).diff(moment(), 'days');
        const weeksRemaining = Math.floor(duration / 7);
        const daysRemaining = duration % 7;
        const currentWeek = 40 - weeksRemaining;
        // console.log('current week ' + currentWeek);
        if (duration <= 0 || weeksRemaining > 40 || duration === undefined || isNaN(duration)) {
            this.setState({
                isBorn: true,
                isReady: true,
            });
        } else {
            this.setState({
                weeksRemaining: weeksRemaining,
                daysRemaining: daysRemaining,
                currentWeek: currentWeek,
                isReady: true,
            });
        }
    }


    render() {
        return (
            <Container style={styles.container}>
                <Header  />
                <View style={{
                    flexDirection: "column",
                    flex: 1,
                    marginBottom: variable.footerHeight
                }}>
                    <Row style={{
                        flex: 1,
                        padding: 0
                    }}>
                        <ImageBackground
                            source={cover}
                            style={[styles.homeImage, styles.bottom]}
                        >
                            <DueDateBanner
                                weeksRemaining={this.state.weeksRemaining}
                                daysRemaining={this.state.daysRemaining}
                                currentWeek={this.state.currentWeek}
                                isBorn={this.state.isBorn}
                                birthday={this.state.birthday}
                            />
                        </ImageBackground>
                    </Row>
                </View>
                <Footer navigation={this.props.navigation}/>
            </Container>
        );
    }
}

export default Home;
