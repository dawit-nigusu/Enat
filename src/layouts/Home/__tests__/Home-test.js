import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from "../index";

it('renders home screen', () => {
    const navigation = { navigate: jest.fn() };
    expect(renderer.create(
        <Home navigation={navigation}/>
    ).toJSON()).toMatchSnapshot();
});
