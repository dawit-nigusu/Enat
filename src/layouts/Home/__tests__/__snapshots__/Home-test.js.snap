// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home screen 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
  virtual={undefined}
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 70,
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 27,
              },
              Object {
                "alignSelf": "center",
              },
              Object {
                "fontFamily": "Icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 27,
              "paddingStart": 5,
            }
          }
        >
          እናት
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 50,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginRight": -9,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 27,
                },
                undefined,
                Object {
                  "fontFamily": "Icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginBottom": 55,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "rgb(194, 24, 91)",
          "flex": 0.2,
          "flexDirection": "row",
          "padding": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 0,
              "width": "100%",
            },
          ]
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "white",
                "fontSize": 80,
                "fontWeight": "bold",
                "textAlign": "center",
                "textAlignVertical": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          እናት
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.8,
          "flexDirection": "row",
          "padding": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "width": null,
            },
            Object {
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "rgba(235, 58, 124, 0.5)",
              "flex": 0,
              "flexDirection": "column",
              "padding": 7,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#ffffff",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            አሁን ያለሽበት ሳምንት
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#ffffff",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            0
             ሳምንት
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#ffffff",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            ልጅሽን ለመውለድ የቀረሽ
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#ffffff",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            0
             ሳምንት ከ
            0
             ቀን
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#cbcbcb",
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 0,
          "right": 0,
        },
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 27,
              },
              undefined,
              Object {
                "fontFamily": "Icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          የቤተሰብ ዕቅድ
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 27,
              },
              undefined,
              Object {
                "fontFamily": "Icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          ላንቺ
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 27,
              },
              undefined,
              Object {
                "fontFamily": "Icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          ለልጅሽ
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 27,
              },
              undefined,
              Object {
                "fontFamily": "Icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          ላጋርሽ
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 27,
              },
              undefined,
              Object {
                "fontFamily": "Icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          ብታውቂው
        </Text>
      </View>
    </View>
  </View>
</View>
`;
